src/app/components/UploadImage/UploadImage.tsx

```diff
@@ -0,0 +1,44 @@
+import styles from './UploadImage.module.css';
+import UploadImageIcon from '../../components/UploadImage/UploadImageIcon.svg';
+import { ChangeEvent, useState } from 'react';
+
+type ImageUploadProps = {
+  onUpload: (url: string) => void;
+};
+
+function UploadImage({ onUpload }: ImageUploadProps): JSX.Element {
+  const [imageSelected, setImageSelected] = useState<File | null>(null);
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (!event.target.files || event.target.files.length === 0) {
+      return;
+    }
+    setImageSelected(event.target.files[0]);
+
+    const formData = new FormData();
+    formData.append('file', imageSelected);
+    formData.append('upload_preset', 'bmnorbj6');
```

Comment:
It would be nice if this token is read from environment variables.
You can add env variables which are readable in client:

https://vitejs.dev/guide/env-and-mode.html#env-variables-and-modes
(don't forget `VITE_` prefix).